resource/resource.go

```diff
@@ -107,7 +107,7 @@ type CheckFailure struct {
 
 type Read interface {
 	Custom
-	Read(ctx Context) error
+	Read(ctx Context, id string) (*pulumirpc.ReadResponse, error)
```

Comment:
I agree that we should change the method to accept the `id`, but the user **should not** need to return a `pulumirpc.ReadResponse`. The library should fabricate a `ReadResponse` by examining the assigned fields of the resource the method was called on, just like `Update` or `Create`.